### PR TITLE
Add neodb extension

### PR DESCRIPTION
**Pull request by sevi418**
## Description
[NeoDB](https://neodb.social/) is an open platform for annotating books, audio, video games and other media in the fediverse.

This extension seamlessly integrates Raycast  and NeoDB, allowing user to easily search NeoDB items and view item details directly within Raycast.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by Sevichecc** [comment]
> Hi @Sevichecc ! Thank you so much for your contribution to Raycast! My name is Milena and I will be reviewing your extension today :)
> 
> So far, it looks good to me! The only thing it concerns me a bit is that there's a bit of delay on the `API` response which is causing a bit of a delay in showing details, for example. Have you considered adding a loading state?

Hi @mil3na ,I have added a loading state using a toast to provide feedback while the API response is pending, what do you think about this approach?

File: extensions/neodb/src/components/BookMeta.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const BookMeta: React.FC<BookMetaProps> = ({ data, rating }) => {
       {translator.length !== 0 && <Detail.Metadata.Label title="Translator" text={translator.join(", ")} />}
       {language && <Detail.Metadata.Label title="Subtitle" text={language} />}
       {pub_house && <Detail.Metadata.Label title="Publication" text={pub_house} />}
-      {pub_year && <Detail.Metadata.Label title="Publication" text={pub_year.toString()} />}
+      {pub_year && <Detail.Metadata.Label title="Published Time" text={pub_year.toString()} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/BookMeta.tsx:21]: maybe just `Published` would be better? `Time` is usually used for more detailed `date`

File: extensions/neodb/src/components/MusicMeta.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          {genre.map((genre) => (
+            <Detail.Metadata.TagList.Item text={genre} color={"#eed535"} key={genre} />
+          ))}
+        </Detail.Metadata.TagList>
+      )}
+      {artist.length !== 0 && <Detail.Metadata.Label title="Artist" text={artist.join(", ")} />}
+      {company.length !== 0 && <Detail.Metadata.Label title="Compony" text={company.join(", ")} />}
+      {duration && <Detail.Metadata.Label title="Duration" text={duration.toString()} />}
+      {release_date && <Detail.Metadata.Label title="Artist" text={release_date.toString()} />}
+      {barcode && <Detail.Metadata.Label title="barcode" text={barcode} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/MusicMeta.tsx:27]: maybe it should be capitalized as the other fields?

File: extensions/neodb/src/components/PerformanceMeta.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      {other_title.length !== 0 && <Detail.Metadata.Label title="Other Title" text={other_title.join(",")} />}
+      {renderGenre(genre)}
+      {rating && <Detail.Metadata.Label title="Rating" text={rating.toString()} />}
+      <Detail.Metadata.Separator />
+      {director.length !== 0 && <Detail.Metadata.Label title="Director" text={director.join(", ")} />}
+      {actor.length !== 0 && <Detail.Metadata.Label title="Actor" text={actor.map((actor) => actor.name).join(", ")} />}
+      {orig_creator.length !== 0 && <Detail.Metadata.Label title="Original Creator" text={orig_creator.join(", ")} />}
+      {playwright.length !== 0 && <Detail.Metadata.Label title="Playwright" text={playwright.join(", ")} />}
+      {composer.length !== 0 && <Detail.Metadata.Label title="Composer" text={composer.join(", ")} />}
+      {crew.length !== 0 && <Detail.Metadata.Label title="Composer" text={composer.join(", ")} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/PerformanceMeta.tsx:41]: Are these both fields for Composer?

File: extensions/neodb/src/components/PerformanceMeta.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      {choreographer.length !== 0 && <Detail.Metadata.Label title="choreographer" text={choreographer.join(", ")} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/PerformanceMeta.tsx:42]: Capitalize this word?

File: extensions/neodb/src/components/TVMeta.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    language,
+    site,
+    season_number,
+    episode_count,
+  } = data as TV;
+
+  return (
+    <Detail.Metadata>
+      {orig_title && <Detail.Metadata.Label title="Original Title" text={orig_title} />}
+      {other_title.length !== 0 && <Detail.Metadata.Label title="Other Title" text={other_title.join(",")} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/TVMeta.tsx:29]: maybe `Alternative Title` suits better?

**Comment by mil3na** [comment]
Thanks for the fast response! The change looks good to me. 
I added a few comments about the categories' output.
Please take a look at them. After that, I think we are good to go!

File: extensions/neodb/src/components/GameMeta.tsx [hunk 6]
```diff
@@ -17,7 +17,7 @@ const GameMeta: React.FC<GameMetaProps> = ({ data, rating }) => {
       {developer.length !== 0 && <Detail.Metadata.Label title="Developer" text={developer.join(", ")} />}
       {publisher.length !== 0 && <Detail.Metadata.Label title="Publisher" text={publisher.join(", ")} />}
       {platform.length !== 0 && <Detail.Metadata.Label title="Platform" text={platform.join(", ")} />}
-      {release_date && <Detail.Metadata.Label title="Release Date" text={release_date.toString()} />}
+      {release_date && <Detail.Metadata.Label title="Released Date" text={release_date.toString()} />}
```
Comment (mil3na) [inline extensions/neodb/src/components/GameMeta.tsx:20]: This should be `Release Date`.

**Comment by mil3na** [comment]
Awesome! Thank you so much! I'm just waiting for the checks to be done and I will be merging soon :)

**Pull request merged**